pkgs/dkg/drand.go

```diff
@@ -159,27 +203,181 @@ func (o *LocalOwner) StartDKG() error {
 	}
 	// New protocol
 	p, err := wire.NewDKGProtocol(&wire.Config{
-		Identifier: o.data.ReqID[:],
-		Secret:     o.data.Secret,
-		Nodes:      nodes,
-		Suite:      o.suite,
-		T:          int(o.data.init.T),
-		Board:      o.b,
-
-		Logger: o.Logger,
+		Identifier: o.Data.ReqID[:],
+		Secret:     o.Data.Secret,
+		NewNodes:   nodes,
+		Suite:      o.Suite,
+		T:          int(o.Data.Init.T),
+		Board:      o.Board,
+		Logger:     o.Logger,
 	})
 	if err != nil {
 		return err
 	}
-
+	// Wait when the protocol exchanges finish and process the result
 	go func(p *dkg.Protocol, postF func(res *dkg.OptionResult) error) {
 		res := <-p.WaitEnd()
 		postF(&res)
 	}(p, o.PostDKG)
-	close(o.startedDKG)
+	close(o.StartedDKG)
+	return nil
+}
+
+func (o *LocalOwner) StartReshareDKGOldNodes() error {
+	o.Logger.Info("Starting Resharing DKG ceremony at old nodes")
+	NewNodes := make([]dkg.Node, 0)
+	for _, op := range o.Data.Reshare.NewOperators {
+		if o.Exchanges[op.ID] == nil {
+			return fmt.Errorf("no operator at Exchanges")
+		}
+		e := o.Exchanges[op.ID]
+		p := o.Suite.G1().Point()
+		if err := p.UnmarshalBinary(e.PK); err != nil {
+			return err
+		}
+
+		NewNodes = append(NewNodes, dkg.Node{
+			Index:  dkg.Index(op.ID - 1),
+			Public: p,
+		})
+	}
```

Comment:
I've seen this same code block at least 3 times, worth having a dedicated function that recvs the operators map `newOperators` in this case and returns the nodes, `newNodes` in this case)